src/Service/ImageHandler.php

```diff
@@ -41,27 +44,42 @@ public function fetch(string $url): string
         }
 
         $response = $this->client->request('GET', $url);
-        if (200 !== $response->getStatusCode()) {
+        if (Response::HTTP_OK !== $response->getStatusCode()) {
             throw new ImageFetchException(sprintf('Failed to fetch %s with code %s', $url, $response->getStatusCode()), $response->getStatusCode());
         }
 
-        $dest = $path.basename($url);
-        $fileHandler = fopen($dest, 'w');
-        foreach ($this->client->stream($response) as $chunk) {
-            fwrite($fileHandler, $chunk->getContent());
+        $headers = $response->getHeaders();
+        $type = reset($headers['content-type']);
+        $dest = $path.$this->generateLocalFilename($url, $type);
+
+        $fetchFile = true;
+        if ($filesystem->exists($dest)) {
+            $size = intval(reset($headers['content-length']));
+            if ($size === filesize($dest)) {
+                // File exist with the same file size.
+                $fetchFile = false;
+            }
+        }
+
+        // Only download files if changes detected in existing file.
+        if ($fetchFile) {
```

Comment:
Well, you actually only check if the size of the file has changed. 

The two images (`docker run -v .:/app dpokidov/imagemagick xc:blue -resize 10x10 /app/blue_10x10.jpg`) ![blue_10x10](https://github.com/itk-dev/event-database-imports/assets/11267554/416fe4eb-976f-42a6-8239-823b4154b2de) and ![red_10x10](https://github.com/itk-dev/event-database-imports/assets/11267554/537301d1-77cd-463a-9236-c042beae3226) have the same size, but are obviously not the same image.

Reply:
The best we can do for now, will look into doing something else if this is a problem later on.